"A Godot Engine build callback failed"

<!-- Please search existing issues for potential duplicates before filing yours:
https://github.com/godotengine/godot/issues?q=is%3Aissue
-->

**Godot version:**
51d7026

**OS/device including version:**
<!-- Specify GPU model and drivers if graphics-related. -->
Manjaro

**Issue description:**
<!-- What happened, and what was expected. -->
Mono build fails using MSBuild with the message (mono 5.20):

```
MSBUILD : error MSB1005: Specify a property and its value.
Switch: /p:GodotDefineConstants=GODOT_X11;GODOT_64;

For switch syntax, type "MSBuild /help"
ERROR: call_build: A Godot Engine build callback failed.
```

I removed mono folders, but it's still happening.

Could be related to #28786.

**Steps to reproduce:**


**Minimal reproduction project:**
<!-- Recommended as it greatly speeds up debugging. Drag and drop a zip archive to upload it. -->

--- a/modules/mono/editor/godotsharp_builds.cpp
+++ b/modules/mono/editor/godotsharp_builds.cpp
@@ -369,7 +369,11 @@ bool GodotSharpBuilds::build_project_blocking(const String &p_config, const Vect
 	MonoBuildInfo build_info(GodotSharpDirs::get_project_sln_path(), p_config);
 
 	// Add Godot defines
+#ifdef WINDOWS_ENABLED
 	String constants = "GodotDefineConstants=\"";
+#else
+	String constants = "GodotDefineConstants=\\\"";
+#endif
 
 	for (int i = 0; i < p_godot_defines.size(); i++) {
 		constants += "GODOT_" + p_godot_defines[i].to_upper().replace("-", "_").replace(" ", "_").replace(";", "_") + ";";
@@ -379,7 +383,11 @@ bool GodotSharpBuilds::build_project_blocking(const String &p_config, const Vect
 	constants += "GODOT_REAL_T_IS_DOUBLE;";
 #endif
 
+#ifdef WINDOWS_ENABLED
 	constants += "\"";
+#else
+	constants += "\\\"";
+#endif
 	build_info.custom_props.push_back(constants);
 
 	if (!GodotSharpBuilds::get_singleton()->build(build_info)) {

